enterprise/server/remote_execution/executor/executor.go

```diff
@@ -236,8 +236,10 @@ func (s *Executor) ExecuteTaskAndStreamResults(ctx context.Context, st *repb.Sch
 		DoNotCache:               task.GetAction().GetDoNotCache(),
 	}
 	finishWithErrFn := func(finalErr error) (retry bool, err error) {
+		retry = false
 		if shouldRetry(task, finalErr) {
-			return true, finalErr
```

Comment:
We flush to clickhouse when sending the COMPLETED event. Currently we're not sending that event for workflow retries. Here's where we change to sending that event

Reply:
Should we make the app changes separately and deploy those changes first, then deploy the executor changes in a separate release?